src/libraries/System.Security.Cryptography.Cng/src/System/Security/Cryptography/AesCng.cs

```diff
@@ -158,14 +161,81 @@ public override void GenerateIV()
                 iv: iv.ToArray(),
                 encrypting: false,
                 paddingMode,
-                CipherMode.CBC);
+                CipherMode.CBC,
+                feedbackSizeInBits: 0);
 
             using (transform)
             {
                 return transform.TransformOneShot(ciphertext, destination, out bytesWritten);
             }
         }
 
+        protected override bool TryDecryptCfbCore(
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> iv,
+            Span<byte> destination,
+            PaddingMode paddingMode,
+            int feedbackSizeInBits,
+            out int bytesWritten)
+        {
+            ValidateCFBFeedbackSize(feedbackSizeInBits);
+
+            UniversalCryptoTransform transform = _core.CreateCryptoTransform(
+                iv: iv.ToArray(),
+                encrypting: false,
+                paddingMode,
+                CipherMode.CFB,
+                feedbackSizeInBits);
+
+            using (transform)
+            {
+                return transform.TransformOneShot(ciphertext, destination, out bytesWritten);
+            }
+        }
+
+        protected override bool TryEncryptCfbCore(
+            ReadOnlySpan<byte> plaintext,
+            ReadOnlySpan<byte> iv,
+            Span<byte> destination,
+            PaddingMode paddingMode,
+            int feedbackSizeInBits,
+            out int bytesWritten)
+        {
+            ValidateCFBFeedbackSize(feedbackSizeInBits);
+
+            UniversalCryptoTransform transform = _core.CreateCryptoTransform(
+                iv: iv.ToArray(),
+                encrypting: true,
+                paddingMode,
+                CipherMode.CFB,
+                feedbackSizeInBits);
+
+            using (transform)
+            {
+                return transform.TransformOneShot(plaintext, destination, out bytesWritten);
+            }
+        }
+
+        private void ValidateCFBFeedbackSize(int feedback)
+        {
+            if (_core.KeyInPlainText)
```

Comment:
This is our detection mechanism to determine if `BasicSymmetricCipherNCrypt` or `BasicSymmetricCipherBCrypt` is going to be used. The former does not correctly handle feedback sizes other than CFB8.